api/proto/metrics/metrics.proto

```diff
@@ -468,6 +468,41 @@ message ZMetricVolume {
   uint64 freeBytes = 9;
 }
 
+// Metrics for a snapshot
+// When a snapshot is created, its disk space is initially shared between
+// the snapshot and the file system, and possibly with previous snapshots.
+// As the file system changes, disk space that was previously shared becomes
+// unique to the snapshot, and thus is counted in the snapshot's used property.
+// Additionally, deleting snapshots can increase the amount of disk space
+// unique to (and thus used by) other snapshots.
+message ZMetricSnapshot {
+  // Snapshot UUID
+  string uuid = 1;
+  // User-friendly name on controller
+  string display_name = 2;
+  // Identifies the amount of space consumed by the dataset and all its
+  // descendants. (in byte)
+  uint64 used_space = 3;
```

Comment:
Is this what is unique to this snapshot i.e., what will be freed up if the snapshot is deleted?
Makes sense to make that clear in the comment.

Reply:
it's unique to this snapshot. I will add a comment

Reply:
I expanded the comment